src/Microsoft.DotNet.ProjectModel.Loader/ProjectLoadContext.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+
+namespace Microsoft.DotNet.ProjectModel.Loader
+{
+    /// <summary>
+    /// An <see cref="AssemblyLoadContext"/> designed to allow loading from a Project.
+    /// </summary>
+    public class ProjectLoadContext : AssemblyLoadContext
+    {
+        private readonly IDictionary<AssemblyName, string> _assemblyPaths;
+        private readonly IDictionary<string, string> _nativeLibraries;
+        private readonly IEnumerable<string> _searchPaths;
+
+        private static readonly string[] NativeLibraryExtensions;
+        private static readonly string[] ManagedAssemblyExtensions = new[]
+        {
+            ".dll",
+            ".ni.dll",
+            ".exe",
+            ".ni.exe"
+        };
+
+
+        static ProjectLoadContext()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                NativeLibraryExtensions = new[] { ".dll" };
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                NativeLibraryExtensions = new[] { ".dylib" };
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                NativeLibraryExtensions = new[] { ".so" };
+            }
+            NativeLibraryExtensions = new string[0];
```

Comment:
Oh, right, heh it's not just messy, it's broken :). It used to be a method that returned a value ;)